src/lib/parameters/flashparams/flashfs32.c

```diff
@@ -0,0 +1,1201 @@
+
+				/* Test the CRC */
+
+				if (pf->crc == crc32(entry_crc_start(pf), entry_crc_length(pf))) {
+
+					/* Valid Magic and CRC look for the next record*/
+
+					pmagic = ((uint32_t *) next_entry(pf));
+
+				} else {
+
+					pmagic++;
+				}
+			}
+
+			if (blank_magic(pmagic)) {
+
+				flash_entry_header_t *pf = (flash_entry_header_t *) pmagic;
+
+				if (blank_entry(pf) && blank_check(pf, required)) {
+					return pf;
+				}
+
+			}
+		}  while (++pmagic != pe);
+	}
+
+	return NULL;
+}
+
+/****************************************************************************
+ * Name: get_next_sector_descriptor
+ *
+ * Description:
+ *   Given a pointer to sector_descriptor_t, this helper function
+ *   returns a pointer to the next sector_descriptor_t
+ *
+ * Input Parameters:
+ *   current      - A pointer to the current sector_descriptor_t
+ *
+ * Returned value:
+ *  On Success A pointer to the next sector_descriptor_t,
+ *  otherwise NULL
+ *
+ *
+ ****************************************************************************/
+
+static sector_descriptor_t *get_next_sector_descriptor(sector_descriptor_t *
+		current)
+{
+	for (int s = 0; sector_map[s].address; s++) {
+		if (current == &sector_map[s]) {
+			if (sector_map[s + 1].address) {
+				s++;
+
+			} else {
+				s = 0;
+			}
+
+			return &sector_map[s];
+		}
+	}
+
+	return NULL;
+}
+
+/****************************************************************************
+ * Name: get_sector_info
+ *
+ * Description:
+ *   Given a pointer to a flash entry header returns the sector descriptor
+ *   for the file is located in
+ *
+ * Input Parameters:
+ *   current      - A pointer to the current flash entry header
+ *
+ * Returned value:
+ *  On Success A pointer to the next sector_descriptor_t,
+ *  otherwise NULL
+ *
+ *
+ ****************************************************************************/
+
+static sector_descriptor_t *get_sector_info(flash_entry_header_t *current)
+{
+	for (int s = 0; sector_map[s].address != 0; s++) {
+		uint8_t *pb = (uint8_t *) sector_map[s].address;
+		uint8_t *pe = pb + sector_map[s].size - 1;
+		uint8_t *pc = (uint8_t *) current;
+
+		if (pc >= pb && pc <= pe) {
+			return &sector_map[s];
+		}
+	}
+
+	return 0;
+}
+
+/****************************************************************************
+ * Name: erase_sector
+ *
+ * Description:
+ *   Given a pointer to sector_descriptor_t, this function
+ *   erases the sector using the pointer to the
+ *   flash_entry_header_t as a sanity check
+ *
+ * Input Parameters:
+ *   sm      - A pointer to the current sector_descriptor_t
+ *   pf      - A pointer to the current flash entry header
+ *
+ * Returned value:
+ *  O On Success or a negative errno
+ *
+ *
+ ****************************************************************************/
+
+static int erase_sector(sector_descriptor_t *sm, flash_entry_header_t *pf)
+{
+	int rv = 0;
+
+	/*
+	* Because up_progmem interface doesn't allow one-to-one check of
```

Comment:
@taylornelms15 - Do you feel strongly that we need this? Have you seen an H7 without all 128K Sectors?

Reply:
The original code had a check for the address being erased being in the requested erase block. This is a way to replicate that check. The uniform memory check is unnecessary if we're stuck on the STM32H7, but porting this to another chip in the future (if someone needs to do so) would lead to issues, presuming they use this method to check that the block and erase address match.
But I don't feel strongly about it either way.